### PR TITLE
Migrate to v2 of GNIP

### PR DESCRIPTION
**Pull request by ethervoid**
Migration to GNIP 2.0 for the twitter search datasource

I will freeze this release until we know something about the GNIP `has:profile_geo` operator.

### CODE REVIEW

**Comment by ethervoid** [comment]
retest this please

**Comment by ethervoid** [comment]
retest this please

**Comment by ethervoid** [comment]
check this please

**Comment by ethervoid** [comment]
@rafatower please CR this :)

File: services/datasources/lib/datasources/search/twitter.rb [hunk 1]
```diff
@@ -320,7 +333,7 @@ def do_search(api_config, redis_storage, filters, stream)
             # might not yet have new value, so introduce a small delay on each thread creation
             sleep(0.1)
             threads[category[CATEGORY_NAME_KEY]] = Thread.new {
-              api = TwitterSearch::SearchAPI.new(api_config, redis_storage, @csv_dumper)
+              api = TwitterSearch::SearchAPI.new(api_config, @user, redis_storage, @csv_dumper)
```
Comment (jgoizueta) [inline services/datasources/lib/datasources/search/twitter.rb]: If `@user` is needed, how was it working before?

Comment (ethervoid) [inline services/datasources/lib/datasources/search/twitter.rb]: I've added the user dependency to the constructor to be able to check for feature flag

**Comment by rafatower** [comment]
:+1:

**Pull request merged**